core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -413,4 +447,56 @@ public InstanceId getInstanceId() {
     return context.getInstanceID();
   }
 
+  @Override
+  public ServerId getServer(ServerTypeName type, String host, int port) {
+    final HostAndPort hp = HostAndPort.fromParts(host, port);
+    switch (type) {
+      case COMPACTOR:
+        return context.getServerIdResolver().resolveCompactor(hp.toString());
+      case MANAGER:
+        final ManagerServerId msi = context.getServerIdResolver().getManager();
+        if (msi.getHost().equals(host) && msi.getPort() == port) {
+          return msi;
+        } else {
+          return null;
+        }
+      case SCAN_SERVER:
+        return context.getServerIdResolver().resolveScanServer(hp.toString());
+      case TABLET_SERVER:
+        return context.getServerIdResolver().resolveTabletServer(hp.toString());
+      default:
+        throw new IllegalArgumentException("Unhandled server type: " + type);
+    }
+  }
+
+  @Override
+  public Set<ServerId> getServers(ServerTypeName type) {
+    return getServers(type, null);
+  }
+
+  @Override
+  public Set<ServerId> getServers(ServerTypeName type, Predicate<ServerId> test) {
+    final Set<ServerId> results = new HashSet<>();
+    switch (type) {
+      case COMPACTOR:
+        results.addAll(context.getServerIdResolver().getCompactors());
+        break;
+      case MANAGER:
+        results.add(context.getServerIdResolver().getManager());
+        break;
+      case SCAN_SERVER:
+        results.addAll(context.getServerIdResolver().getScanServers().keySet());
+        break;
+      case TABLET_SERVER:
+        results.addAll(context.getServerIdResolver().getTabletServers());
+        break;
+      default:
+        break;
+    }
+    if (test == null) {
+      return results;
```

Comment:
Could avoid returning mutable sets.

```suggestion
      return Collections.unmodifiableSet(results);
```

For the stream I think Collectors has a function for collecting to an unmodifiable set.

Reply:
Suggestion implemented in ebd6763